File: JavaScript/48-Rotate Image.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */

var rotate = function(matrix) {
    let N = matrix.length;
    
    for (let i=0; i<N; i++){
        for (let j=0; j<i; j++){
            let tmp = matrix[i][j];
            matrix[i][j] = matrix[j][i]; 
            matrix[j][i] = tmp;
        }
    }
    
    for (let row of matrix){
        row = row.reverse();
    }
};

File: C++/48-Rotate Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int row = matrix.size(), col = matrix[0].size();

        for (int r=1; r<row; r++) {
            for (int c=0; c<r; c++) {
                swap(matrix[r][c], matrix[c][r]);
            }
        }

        for (auto &r:matrix) {
            reverse(r.begin(), r.end());
        }

    }
};